packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -2935,8 +2936,8 @@ export class ThinEngine {
         indexParameters?: any,
         shaderLanguage = ShaderLanguage.GLSL
     ): Effect {
-        const vertex = baseName.vertexElement || baseName.vertex || baseName.vertexToken || baseName.vertexSource || baseName;
-        const fragment = baseName.fragmentElement || baseName.fragment || baseName.fragmentToken || baseName.fragmentSource || baseName;
+        const vertex = typeof baseName === "string" ? baseName : baseName.vertexSource || baseName.vertexElement || baseName.vertex || (baseName as any).vertexToken;

```

Comment:
You should keep the same order than previously: `baseName.vertexElement || baseName.vertex || baseName.vertexToken || baseName.vertexSource`.

Same thing for the `fragment` case.

Reply:
I intentionally changed this, so that it matches with the order in `Materials/effect.ts`. Is that alright?
(I also politely added that priority info to the IShaderPath docs.)

Reply:
Ok for me!

The "tokenXXX" check should come first.

Reply:
Fixed :)